sycl/doc/extensions/experimental/sycl_ext_oneapi_bindless_images/ZE_experimental_bindless_images.asciidoc

```diff
@@ -0,0 +1,621 @@
+/// @brief Describes memory to be allocated on device intended for images with
+///        a device specific memory layout
+///
+/// @details
+///     - This structure is passed to:
+///       + ::zeMemAllocImageExp
+///       + ::ze_image_desc_t.pNext
+///         - When this structure is passed to an image descriptor,
+///           ::zeImageCreate must not allocate any additional device memory for
+///           the created image, only return a handle.
+///         - Additionally, when ::zeImageDestroy is called on an image created
+///           from a ::ze_image_mem_alloc_exp_desc_t allocation,
+///           ::zeImageDestroy must not free the memory associated with the
+///           image handle.
+///
+typedef struct _ze_image_mem_alloc_exp_desc_t {
+    ze_structure_type_t stype;      // [in]
+    void *pNext;                    // [in,out][optional]
+    size_t widthInBytes;            // [in]
+    size_t height;                  // [in][optional]
+    size_t depth;                   // [in][optional]
+    uint32_t elementSizeByte;       // [in]
+} ze_image_mem_alloc_exp_desc_t;
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Describes linear USM to be allocated on device intended for images
+///
+/// @details
+///     - This structure is passed to:
+///       + ::zeMemAllocPitchedExp
+///         - ::rowPitch and ::slicePitch are [out] parameters
+///       + ::ze_image_desc_t.pNext
+///         - ::rowPitch and ::slicePitch are [in] parameters
+///         - When this structure is passed to an image descriptor,
+///           ::zeImageCreate must not allocate any additional device memory for
+///           the created image, only return a handle.
+///         - Additionally, when ::zeImageDestroy is called on an image created
+///           from a ::ze_image_mem_alloc_exp_desc_t allocation,
+///           ::zeImageDestroy must not free the memory associated with the
+///           image handle.
+///
+typedef struct _ze_image_usm_alloc_exp_desc_t {
+    ze_structure_type_t stype;      // [in]
+    void *pNext;                    // [in,out][optional]
+    size_t widthInBytes;            // [in]
+    size_t height;                  // [in][optional]
+    size_t depth;                   // [in][optional]
+    uint32_t elementSizeBytes;      // [in]
+    size_t rowPitch;                // [in,out]
+    size_t slicePitch;              // [in,out]
+} ze_image_usm_alloc_exp_desc_t;
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Describes device specific restrictions for image properties
+///
+/// @details
+///     - This structure is passed to:
+///       + ::ze_device_properties_t::pNext
+///
+typedef struct _ze_device_image_bindless_exp_desc_t {
+    ze_structure_type_t stype;      // [in]
+    void *pNext;                    // [in,out][optional]
+    uint32_t imagePitchAlign;       // [out]
```

Comment:
imagePitchAlign -> minimum Alignment I guess?

Reply:
is it only row pitch aligment ?
how about 3D images that may have slice pitch alignment ?

Reply:
Answered above. This will need to be extended to support 3D images fully